lib/puppet/indirector/request.rb

```diff
@@ -196,26 +196,39 @@ def do_request(srv_service=:puppet, default_server=nil, default_port=nil, &block
       end
     end
 
-    # ... Fall back onto the default server.
-     bound_server = Puppet.lookup(:server) do
-      if primary_server = Puppet.settings[:server_list][0]
-        primary_server[0]
-      else
-        Puppet.settings[:server]
+    if default_server
+      self.server = default_server
+      Puppet.debug _("Selected default server: %{server}") % {server: default_server}
```

Comment:
It looks like these default debugs are redundant and likely unnecessary

@jhelwig

Reply:
It looks like the only this that calls this method (`do_request`) is the `do_request` method in `lib/puppet/indirector/rest.rb`, which is where we have the `dynamically bound lookup failed` debug messages

```
$ git grep do_request lib
lib/puppet/indirector/facts/rest.rb:    response = do_request(request) do |req|
lib/puppet/indirector/request.rb:  def do_request(srv_service=:puppet, default_server=nil, default_port=nil, &block)
lib/puppet/indirector/rest.rb:    response = do_request(request) do |req|
lib/puppet/indirector/rest.rb:    response = do_request(request) do |req|
lib/puppet/indirector/rest.rb:    response = do_request(request) do |req|
lib/puppet/indirector/rest.rb:    response = do_request(request) do |req|
lib/puppet/indirector/rest.rb:    response = do_request(request) do |req|
lib/puppet/indirector/rest.rb:  # Encapsulate call to request.do_request with the arguments from this class
lib/puppet/indirector/rest.rb:  # We certainly could have retained the full request.do_request(...) { |r| ... }
lib/puppet/indirector/rest.rb:  # to request.do_request from here, thus if we change what we pass or how we
lib/puppet/indirector/rest.rb:  def do_request(request)
lib/puppet/indirector/rest.rb:    response = request.do_request(self.class.srv_service, self.class.server, self.class.port) { |req| yield(req) }
lib/puppet/network/http/connection.rb:      do_request(Net::HTTP::Get.new(path, headers), options)
lib/puppet/network/http/connection.rb:      do_request(request, options)
lib/puppet/network/http/connection.rb:      do_request(Net::HTTP::Head.new(path, headers), options)
lib/puppet/network/http/connection.rb:      do_request(Net::HTTP::Delete.new(path, headers), options)
lib/puppet/network/http/connection.rb:      do_request(request, options)
lib/puppet/network/http/connection.rb:    def do_request(request, options)
lib/puppet/type/file/source.rb:      request.do_request(:fileserver) do |req|
```

Reply:
I removed it in an additional commit, which should be squashed before this is merged